x/evm/types/state_transition.go

```diff
@@ -109,6 +109,28 @@ func (st StateTransition) newEVM(
 	return vm.NewEVM(blockCtx, txCtx, csdb, config.EthereumConfig(st.ChainID), vmConfig)
 }
 
+func (st StateTransition) refundGas(gasInfo GasInfo) {
+	// Apply refund counter, capped to half of the used gas.
+	refund := gasInfo.GasConsumed / 2
+	if refund > gasInfo.GasRefunded {
+		refund = gasInfo.GasRefunded
+	}
+
+	// Return ETH for remaining gas, exchanged at the original rate.
+	gasRefund := new(big.Int).Mul(new(big.Int).SetUint64(refund), st.Price)
+
+	// add refund back to counter.
+	// add gas information into state db
+	st.Csdb.AddRefund(gasRefund.Uint64())
```

Comment:
seems `CommitStateDB` method  `AddRefund` takes one param which is `gas`, not refund amount.

Reply:
I don't quite understand what you mean... so youre suggesting that we `AddRefund` the gasConsumed amount then? Do you think you can explain what you mean here